waves/src/components/AssetSelector.tsx

```diff
@@ -0,0 +1,108 @@
+import { Center, NumberInput, NumberInputField, VStack } from "@chakra-ui/react";
+import React from "react";
+import { AssetType } from "../App";
+import AssetSelect from "./AssetSelect";
+
+interface AssetSelectorProps {
+    type: AssetType;
+    amount: number;
+    placement: "left" | "right";
+    onTypeChange: (asset: AssetType) => void;
+    onAmountChange: (asset: number) => void;
+}
+
+function AssetSelector({ type, amount, onTypeChange, onAmountChange, placement }: AssetSelectorProps) {
+    const box_width = 400;
+    const box_height = 220;
+
+    return (
+        <Center bg="gray.100" w={box_width} h={box_height} borderRadius={"md"}>
+            <VStack spacing={4} id="select{type}">
+                <AssetSelect type={type} onAssetChange={onTypeChange} placement={placement} />
+                {/* asset is BTC: render BTC input*/}
+                {type === AssetType.BTC
+                    && <BitcoinInput amount={amount} onAmountChange={onAmountChange} />}
+                {/* asset is USDT: render USDT input*/}
+                {type === AssetType.USDT
+                    && <UsdtInput amount={amount} onAmountChange={onAmountChange} />}
+            </VStack>
+        </Center>
+    );
+}
+
+export default AssetSelector;
+
+interface BitcoinInputProps {
+    amount: number;
+    onAmountChange: (amount: number) => void;
+}
+
+function BitcoinInput({ amount, onAmountChange }: BitcoinInputProps) {
+    const format = (val: number) => {
+        return `₿ ` + val;
+    };
+
+    const parse = (val: string) => {
+        return Number(val.replace(/^₿/, ""));
+    };
+
+    const updateValue = (val: string) => {
+        let updatedValue = parse(val);
+        onAmountChange(updatedValue);
+    };
+
+    return (
+        <CustomInput value={amount} precision={8} step={0.00000001} updateValue={updateValue} format={format} />
+    );
+}
+
+interface UsdtInputProps {
+    amount: number;
+    onAmountChange: (amount: number) => void;
+}
+
+function UsdtInput({ amount, onAmountChange }: UsdtInputProps) {
+    const format = (val: number) => {
+        return `$ ` + val;
+    };
+
+    const parse = (val: string) => {
+        return Number(val.replace(/^\$/, ""));
+    };
+
+    const updateValue = (val: string) => {
+        let updatedValue = parse(val);
+        onAmountChange(updatedValue);
+    };
+
+    return (
+        <CustomInput value={amount} precision={2} step={0.01} updateValue={updateValue} format={format} />
+    );
+}
+
+interface CustomInputProps {
+    value: number;
+    precision: number;
+    step: number;
+    updateValue: (val: string) => void;
+    format: (val: number) => string;
+}
+
+function CustomInput({ value, updateValue, precision, step, format }: CustomInputProps) {
```

Comment:
To avoid low-quality names like this, I'd usually just import the original one under a different name and name my own component `NumberInput`.

You can do:

```ts
import { Center, NumberInput as CUINumberInput, NumberInputField, VStack } from "@chakra-ui/react";
```

With `CUI` meaning "Chakra-UI".

A different approach would be to ask, whether or not you actually want to extract a component here. This component is not doing anything other than setting a couple of props to hardcoded values. You can achieve the same thing by having these props in a constant:

```ts
const ASSET_INPUT_PROPS = {
	size: "lg",
	textStyle: "actionable"
};

function UsdtInput({ amount, onAmountChange }: UsdtInputProps) {
    const format = (val: number) => {
        return `$ ` + val;
    };

    const parse = (val: string) => {
        return Number(val.replace(/^\$/, ""));
    };

    const updateValue = (val: string) => {
        let updatedValue = parse(val);
        onAmountChange(updatedValue);
    };

    return (
        <NumberInput {...ASSET_INPUT_PROPS} value={amount} precision={2} step={0.01} updateValue={updateValue} format={format} />
    );
}
```

Be mindful of not creating too many nested levels of components. A flatter hierarchy is usually better!